phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/DefaultPhoenixMultiViewListProvider.java

```diff
@@ -0,0 +1,141 @@
+                            // CASE 1 : BASE_TABLE -> GLOBAL_VIEW -> TENANT_VIEW
+                            // CASE 2 : BASE_TABLE -> TENANT_VIEW
+                            // CASE 2 : BASE_TABLE -> VIEW
+                            PTable parentTable = PhoenixRuntime.getTable(connection, null,
+                                    pTable.getParentName().toString());
+                            if (parentTable.getType() == PTableType.VIEW &&
+                                    parentTable.getPhoenixTTL() > 0) {
+                                skip = true;
+                            }
+                        } catch (Exception e) {
+                            skip = true;
+                            LOGGER.error(String.format("Had an issue to process the view: %s, tenantId:" +
+                                    "see error %s ", fullTableName, tenantId, e.getMessage()));
+                        }
+
+                        if (!skip) {
+                            ViewInfoWritable viewInfoTracker = new ViewInfoTracker(
+                                    tenantId,
+                                    fullTableName,
+                                    viewTtlValue,
+                                    pTable.getPhysicalName().getString(),
+                                    false
+
+                            );
+                            viewInfoWritables.add(viewInfoTracker);
+
+                            List<PTable> allIndexesOnView = pTable.getIndexes();
+                            for (PTable viewIndexTable : allIndexesOnView) {
+                                String indexName = viewIndexTable.getTableName().getString();
+                                String indexSchema = viewIndexTable.getSchemaName().getString();
+                                if (indexName.contains(QueryConstants.CHILD_VIEW_INDEX_NAME_SEPARATOR)) {
+                                    indexName = SchemaUtil.getTableNameFromFullName(indexName,
+                                            QueryConstants.CHILD_VIEW_INDEX_NAME_SEPARATOR);
+                                }
+                                indexName = SchemaUtil.getTableNameFromFullName(indexName);
+                                indexName = SchemaUtil.getTableName(indexSchema, indexName);
+                                ViewInfoWritable viewInfoTrackerForIndexEntry = new ViewInfoTracker(
+                                        tenantId,
+                                        fullTableName,
+                                        viewTtlValue,
+                                        indexName,
+                                        true
+
+                                );
+                                viewInfoWritables.add(viewInfoTrackerForIndexEntry);
+                            }
+                        }
+                    }
+                    if (isQueryMore) {
+                        if (fullTableName == null) {
```

Comment:
fullTableName is not being used anywhere after this, so why the check?

Reply:
the fullTableName is checking whether we done the full scan of syscat or not.
If we do have more rows to query, the current fullTableName cannot be null right? If we don't have more rows to scan, the current rs.next will not have any value, and we reset the fullTableName value to null for every iteration.